public/app/plugins/datasource/graphite/specs/datasource.test.ts

```diff
@@ -303,6 +303,17 @@ describe('graphiteDatasource', () => {
       expect(requestOptions.params.expr).toEqual(['server=~backend*']);
       expect(results).not.toBe(null);
     });
+
+    it('should not be canceled for long queries', () => {
+      const longQuery = 'a'.repeat(40000);
```

Comment:
Not sure this is needed since it's more of a browser/server limitation, right? But the rest of the test make sense. If you want to make sure that referencing template variables are interpolated (see other comment) that could also be added to this test.